napari/_qt/qt_main_window.py

```diff
@@ -298,6 +299,11 @@ def __init__(self, viewer, *, show: bool = True):
         else:
             self._debug_menu = None
 
+        for name, action in action_manager._actions.items():
+            qa = QAction(self.qt_viewer)
+            action_manager.bind_qaction(name, qa)
+            self.window_menu.addAction(qa)
+
```

Comment:
This currently add all existing action to the view menu I'm not sure we want this.

Reply:
How do we handle localizationof actions If `name`  is acting as a unique `id?

Reply:
Ah, good question.
We can also not localize the name as the description is localized, and rename name to ID. right now the name is also used in menus, so we can add one short field.

Reply:
Yes, lets use `id_` (ir similar) for uniqueness and `name` for menus (which can be localized)

Reply:
> yeah, I don't think the window menu is the right place for this. I do kinda like having them all in a menu somewhere, but it might need a new menu, (Actions?). Also, I think the benefit of having them in a menu somewhere would be increased quite a bit if they had the keyboard shortcut next the the action:

Yeah, I had it but removed it for the following reason:
 - Adding a shortcut to a qaction is not purely visual it also binds the shortcut at the window level.

it was thus bypassing the KeymapProvider. It is fine for global actions like "Reset Viewer", but does not work for other that needs an instance (like button in layers). It is fixable by adding a notion of active instance that should receive the shortcut; but it was easier to remove it for now.

> but we can probably do better with some function to convert our shortcuts to the necessary QAction symbols.

I think internally shortcut should opaque objects, with attributes/properties to get corresponding values in given context. 
We could even not use strings and have something like `sht.CTRL.SHIFT.T` and fail early.

Reply:
We're going to have the same question with layers if we want recording, so yeah, somethign to solve.